scipy/special/cython_special.pxd

```diff
@@ -208,8 +208,8 @@ cpdef double radian(double x0, double x1, double x2) nogil
 cpdef double rel_entr(double x0, double x1) nogil
 cpdef Dd_number_t rgamma(Dd_number_t x0) nogil
 cpdef double round(double x0) nogil
-cdef void shichi(double x0, double *y0, double *y1) nogil
-cdef void sici(double x0, double *y0, double *y1) nogil
+cdef void shichi(Dd_number_t x0, Dd_number_t *y0, Dd_number_t *y1) nogil
+cdef void sici(Dd_number_t x0, Dd_number_t *y0, Dd_number_t *y1) nogil
```

Comment:
Not related to this PR, but to gh-6195 --- it needs to be understood what problems this sort of signature changes can cause to existing Cython code. I think this is similar to an ABI break --- changing signatures here will require recompilation of Cython modules using these exported functions.